docs/01_nodeos/02_usage/05_nodeos-implementation.md

```diff
@@ -10,17 +10,22 @@ Every `nodeos` instance creates some internal files to housekeep the blockchain
 
 * The `block.log` is an append only log of blocks written to disk and contains all the irreversible blocks.
 * `Reversible_blocks` is a memory mapped file and contains blocks that have been written to the blockchain but have not yet become irreversible.
-* The `chain state` or `database` is a memory mapped file, storing the blockchain state of each block (account details, deferred transactions, transactions, data stored using multi index tables in smart contracts, etc.). Once a block becomes irreversible we no longer cache the chain state.
+* The `chain state` or `chain database` is a memory mapped file, storing the blockchain state of each block (account details, deferred transactions, transactions, data stored using multi index tables in smart contracts, etc.). Once a block becomes irreversible, the chain state is no longer cached.
```

Comment:
That `database` or `chain database` or `chain state` is currently cached in that it is held in memory mapped file. All this data is always in the memory mapped file. This includes account details, deferred transactions, and data stored using multi index tables in smart contracts. (Note in the future we will have key-value storage via rocksdb which will not be in memory but on disk). The last 65536 Block Ids are cached to support TaPOS. Transaction id/expiration is cached until the transaction expires.

Outside of chain state, just in normal ram memory, block data is cached for awhile and then is no longer available directly once irreversible is reached. Specifically the signed_block itself. After LIB it is retrieved from the block log.